src/solver/optimisation/variables/VariableManagement.h

```diff
@@ -0,0 +1,171 @@
+    {
+        return nativeOptimVar_
+          .NumeroDeVariableDuNombreDeGroupesQuiSArretentDuPalierThermique[index];
+    }
+
+    int& NumberStartingDispatchableUnits(unsigned int index)
+    {
+        return nativeOptimVar_
+          .NumeroDeVariableDuNombreDeGroupesQuiDemarrentDuPalierThermique[index];
+    }
+
+    int& NumberBreakingDownDispatchableUnits(unsigned int index)
+    {
+        return nativeOptimVar_
+          .NumeroDeVariableDuNombreDeGroupesQuiTombentEnPanneDuPalierThermique[index];
+    }
+
+    int& NTCDirect(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariableDeLInterconnexion[index];
+    }
+
+    int& IntercoDirectCost(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariableCoutOrigineVersExtremiteDeLInterconnexion[index];
+    }
+
+    int& IntercoIndirectCost(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariableCoutExtremiteVersOrigineDeLInterconnexion[index];
+    }
+
+    int& ShortTermStorageInjection(unsigned int index)
+    {
+        return nativeOptimVar_.SIM_ShortTermStorage.InjectionVariable[index];
+    }
+
+    int& ShortTermStorageWithdrawal(unsigned int index)
+    {
+        return nativeOptimVar_.SIM_ShortTermStorage.WithdrawalVariable[index];
+    }
+
+    int& ShortTermStorageLevel(unsigned int index)
+    {
+        return nativeOptimVar_.SIM_ShortTermStorage.LevelVariable[index];
+    }
+
+    int& HydProd(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariablesDeLaProdHyd[index];
+    }
+
+    int& HydProdDown(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariablesVariationHydALaBaisse[index];
+    }
+
+    int& HydProdUp(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariablesVariationHydALaHausse[index];
+    }
+
+    int& Pumping(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariablesDePompage[index];
+    }
+
+    int& HydroLevel(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariablesDeNiveau[index];
+    }
+
+    int& Overflow(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariablesDeDebordement[index];
+    }
+
+    int& FinalStorage(unsigned int index)
+    {
+        return NumeroDeVariableStockFinal[index];
+    }
+
+    int& LayerStorage(unsigned area, unsigned layer)
+    {
+        return NumeroDeVariableDeTrancheDeStock[area][layer];
+    }
+
+    int& PositiveUnsuppliedEnergy(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariableDefaillancePositive[index];
+    }
+
+    int& NegativeUnsuppliedEnergy(unsigned int index)
+    {
+        return nativeOptimVar_.NumeroDeVariableDefaillanceNegative[index];
+    }
+
+private:
+    CORRESPONDANCES_DES_VARIABLES& nativeOptimVar_;
+    std::vector<int>& NumeroDeVariableStockFinal;
+    std::vector<std::vector<int>>& NumeroDeVariableDeTrancheDeStock;
+};
+
+class VariableManagerFactory
```

Comment:
I'm not a big fan of the name of this class. The name "Factory" wrongly suggests that it implements the Factory design pattern. But you only have on VariableManager implementation.

Reply:
Thank you.
I appreciate

Reply:
I need your lights

Reply:
More

Reply:
NOT FOR THIS!